main/file.c

```diff
@@ -77,16 +83,16 @@ sub_status_t sub_init(sub_init_struct_t *init_struct, uint16_t transmit_periods[
 
     if(init_struct->_on_data_rx == NULL){
 
-        sub_struct.rx_data = xQueueCreate(10, sizeof(uint32_t));
-        if(sub_struct.rx_data == NULL){
+        sub_struct.rx_queue = xQueueCreate(10, sizeof(uint32_t));
```

Comment:
Nie zauważyłem tego wcześniej, dlaczego tutaj jest rozmiar uint32_t?

Reply:
zapewne dlatego że wstawiłem jakiś bo nie byłem pewien jaki

Reply:
A masz  pomysł na co to zamienić?

Reply:
jeżeli mogą być dwa typy wiadomości, to może strukturę w której trzymam recived info, bo tam mam pointer na dane i ich długość

Reply:
Z pointerem na dane musisz uważać, żeby task nie nadpisał tych danych na które wskazuje. Trochę to zależy jak zakodujesz task, ale najbezpieczniej tutaj przesyłać kopie danych. Przy okazji możemy zmniejszyć rozmiar tablicy, bo mamy (stałą ramkę komend które przesyłamy i to jest maksymalna wielkość jaką powinniśmy odebrać)[https://github.com/PWrInSpace/MCB_Aurora/blob/4280a27aaaf51b2833eed5d7e26356b08fe27241/components/commands/commands.h#L37-L44].